commands/user.go

```diff
@@ -315,3 +329,45 @@ auth_service: {{.AuthService}}`
 
 	return nil
 }
+
+func listUsersCmdF(c client.Client, command *cobra.Command, args []string) error {
+	// TODO What to do with err?
+	page, errp := command.Flags().GetInt("page")
+	if errp != nil {
+		return errors.Wrap(errp, "Could not parse --page")
+	}
+	perPage, errpp := command.Flags().GetInt("per-page")
+	if errpp != nil {
+		return errors.Wrap(errp, "Could not parse --per-page")
+	}
+	showAll, errs := command.Flags().GetBool("all")
+	if errs != nil {
+		return errors.Wrap(errs, "Could not parse --all")
+	}
+
+	if showAll {
+		page = 0
+	}
+
+	tpl := `{{.Id}}: {{.Username}} ({{.Email}})`
+
+	users, response := c.GetUsers(page, perPage, "")
```

Comment:
I still don't understand why we are calling this twice ? Can't we have a single loop where we are calling this once ?

Reply:
The condition in for loop is:
```users != nil && len(users) > 0```
If server returns error on first call, `users` will be `nil`. So the part inside loop will never execute and `nil` will be returned.

One option is to check for response.Error before [return statement](https://github.com/mattermost/mmctl/pull/34/files#diff-dbf54b6a188b305f65470b9f52908c27R372).

Reply:
Apologies for the confusion. By "calling this twice", I meant why are we writing the code invoking `GetUsers` twice. I get that the method actually gets invoked once, but why do we need to write it twice ? Can't we just `c.GetUsers` once in the code ?

Is that clear ?